api/src/main/java/jakarta/data/restrict/Restriction.java

```diff
@@ -17,6 +17,131 @@
+ * {@linkplain Constraint constraint} are obtained from {@linkplain Attribute}
+ * subtypes of the {@linkplain StaticMetamodel static metamodel}.</p>
+ *
+ * <p>Composite restrictions that impose multiple
+ * {@linkplain Constraint constraints} are obtained from the
+ * {@link Restrict#all(Restriction...)} and
+ * {@link Restrict#any(Restriction...)} methods.</p>
+ *
+ * <p>A repository {@link Find} method can optionally accept a parameter of
+ * type {@code Restriction}. For example,</p>
+ * <pre>
+ * &#64;Repository
+ * public interface Cars extends CrudRepository&lt;Car, String&gt; {
+ *
+ *     &#64;Find
+ *     List&lt;Car&gt; search(&#64;By(_Car.MAKE) String manufacturer,
+ *                      &#64;By(_Car.MODEL) String model,
+ *                      Restriction&lt;Car&gt; restriction,
+ *                      Order&lt;Car&gt;... sort);
+ * }
+ * </pre>
+ *
+ * <p>Instances of restriction obtained from the static metamodel or the
+ * {@link Restrict} class can be supplied as the parameter when invoking the
+ * respository method. For example,</p>
+ * <pre>
+ * List&lt;Car&gt; withinPriceRange =
+ *         cars.search(make,
+ *                     model,
+ *                     _Car.price.between(20000, 30000),
+ *                     Order.by(_Car.price.desc()));
+ *
+ * List&lt;Car&gt; pricedUnder30kWhenDiscounted =
+ *         cars.search(make,
+ *                     model,
+ *                     _Car.price.minus(discount).lessThan(30000),
+ *                     Order.by(_Car.price.desc()));
+ *
+ * List&lt;Car&gt; atLeast2020AndPricedBelow30k =
+ *         cars.search(make,
+ *                     model,
+ *                     Restrict.all(_Car.year.greaterThanEqual(2020),
+ *                                  _Car.price.lessThan(30000)),
+ *                     Order.by(_Car.price.desc()));
+ * </pre>
+ *
+ * <p>Entity class for the above example:</p>
+ * <pre>
+ * &#64;Entity
+ * public class Car {
+ *     &#64;Id
+ *     public String vin;
+ *     public String make;
+ *     public String model;
+ *     public int price;
+ *     public int year;
+ * }
+ * </pre>
+ *
+ * <p>Static metamodel (typically generated from the entity class) for the above entity:</p>
+ * <pre>
+ * &#64;StaticMetamodel
+ * public interface _Car {
+ *     String MAKE = "make";
+ *     String MODEL = "model";
+ *     String PRICE = "price";
+ *     String VIN = "vin";
+ *     String YEAR = "year";
+ *
+ *     TextAttribute&lt;Car&gt; make = TextAttribute.of(
+ *             Car.class, MAKE);
+ *     TextAttribute&lt;Car&gt; model = TextAttribute.of(
+ *             Car.class, MODEL);
+ *     NumericAttribute&lt;Car,Integer&gt; price = NumericAttribute.of(
+ *             Car.class, PRICE. int.class);
+ *     NumericAttribute&lt;Car,Integer&gt; year = NumericAttribute.of(
+ *             Car.class, YEAR. int.class);
+ *     TextAttribute&lt;Car&gt; vin = TextAttribute.of(
+ *             Car.class, VIN);
+ * }
+ * </pre>
```

Comment:
We already have a code example like this on `@StaticMetamodel`. I would not repeat it here. If you want to repeat it somewhere, `Attribute` would be a better place IMO.

Reply:
Yeah, that's probably a good idea. I moved it to `Attribute` as you suggested, and we can just refer to that when we have examples like this that make non-trivial use of the metamodel.